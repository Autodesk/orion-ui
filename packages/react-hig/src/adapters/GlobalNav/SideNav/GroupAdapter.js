/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../createComponent';
import HIGElement from '../../../elements/HIGElement';
import HIGNodeList from '../../../elements/HIGNodeList';
import HIGChildValidator from '../../../elements/HIGChildValidator';
import ModuleComponent, {
  Module
} from '../../../adapters/GlobalNav/SideNav/ModuleAdapter';

export class GroupAdapter extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.modules = new HIGNodeList({
      type: Module,
      HIGConstructor: this.hig.partials.Module,
      onAdd: (instance, beforeInstance) => {
        this.hig.addModule(instance, beforeInstance);
      }
    });
  }

  componentDidMount() {
    this.modules.componentDidMount();
  }

  createElement(ElementConstructor, props) {
    return this.modules.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.modules.insertBefore(instance, insertBeforeIndex);
  }

  removeChild(instance) {
    this.modules.removeChild(instance);
  }
}

const GroupComponent = createComponent(GroupAdapter);

GroupComponent.propTypes = {
  children: HIGChildValidator([ModuleComponent])
};

GroupComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding Module'
    }
  }
};

GroupComponent.Module = ModuleComponent;

export default GroupComponent;
