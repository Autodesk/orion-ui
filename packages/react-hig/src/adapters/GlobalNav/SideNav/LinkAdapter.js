/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import createComponent from '../../createComponent';
import HIGElement from '../../../elements/HIGElement';

export class Link extends HIGElement {
  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'title': {
          this.hig.setTitle(propValue);
          break;
        }
        case 'link': {
          this.hig.setLink(propValue);
          break;
        }
        case 'onClick': {
          const dispose = this._disposeFunctions.get('onClickDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onClickDispose',
            this.hig.onClick(propValue)
          );
          break;
        }
        case 'onHover': {
          const dispose = this._disposeFunctions.get('onHoverDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onHoverDispose',
            this.hig.onHover(propValue)
          );
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const LinkComponent = createComponent(Link);

LinkComponent.propTypes = {
  addTitle: PropTypes.func,
  addLink: PropTypes.func,
  onClick: PropTypes.func,
  onHover: PropTypes.func
};

LinkComponent.__docgenInfo = {
  props: {
    title: {
      description: 'Title {string} for the Link component'
    },
    link: {
      description: 'URL {string} for the Link component'
    },
    onClick: {
      description: '{function} Triggered when user clicks on the link'
    },
    onHover: {
      description: '{function} Triggered when user hovers over the link'
    }
  }
};

export default LinkComponent;
