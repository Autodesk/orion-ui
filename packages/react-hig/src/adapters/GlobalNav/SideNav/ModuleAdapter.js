/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import createComponent from '../../createComponent';
import HIGElement from '../../../elements/HIGElement';
import HIGNodeList from '../../../elements/HIGNodeList';
import HIGChildValidator from '../../../elements/HIGChildValidator';
import SubmoduleComponent, {
  SubmoduleAdapter
} from '../../../adapters/GlobalNav/SideNav/SubmoduleAdapter';
import ModuleCollapseComponent, {
  ModuleCollapseAdapter
} from './ModuleCollapseAdapter';

export class Module extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.submodules = new HIGNodeList({
      type: SubmoduleAdapter,
      HIGConstructor: this.hig.partials.Submodule,
      onAdd: (instance, beforeInstance) => {
        this.hig.addSubmodule(instance, beforeInstance);
      }
    });
  }

  componentDidMount() {
    this.submodules.componentDidMount();

    if (this.collapse) {
      this.hig.addCollapse(this.collapse.hig);
      this.collapse.mount();
    }
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'icon': {
          this.hig.setIcon(propValue);
          break;
        }
        case 'title': {
          this.hig.setTitle(propValue);
          break;
        }
        case 'link': {
          this.hig.setLink(propValue);
          break;
        }
        case 'active': {
          propValue ? this.hig.activate() : this.hig.deactivate();
          break;
        }
        case 'onClick': {
          const dispose = this._disposeFunctions.get('onClickDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onClickDispose',
            this.hig.onClick(propValue)
          );
          break;
        }
        case 'onHover': {
          const dispose = this._disposeFunctions.get('onHoverDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onHoverDispose',
            this.hig.onHover(propValue)
          );
          break;
        }
        case 'children': {
          // No-op
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case SubmoduleAdapter:
        return this.submodules.createElement(ElementConstructor, props);
      case ModuleCollapseAdapter:
        return new ModuleCollapseAdapter(props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  insertBefore(instance, insertBeforeIndex) {
    if (instance instanceof SubmoduleAdapter) {
      this.submodules.insertBefore(instance, insertBeforeIndex);
    } else if (instance instanceof ModuleCollapseAdapter) {
      if (this.collapse) {
        throw new Error('only one ModuleCollapse is allowed');
      } else {
        this.collapse = instance;

        if (this.mounted) {
          instance.componentDidMount();
        }
      }
    } else {
      throw new Error('unknown type');
    }
  }

  appendChild(instance) {
    this.insertBefore(instance);
  }

  removeChild(instance) {
    if (instance instanceof SubmoduleAdapter) {
      this.submodules.removeChild(instance);
    }

    if (instance instanceof ModuleCollapseAdapter) {
      this.collapse = null;
      instance.unmount();
    }
  }
}

const ModuleComponent = createComponent(Module);

ModuleComponent.propTypes = {
  icon: PropTypes.string,
  title: PropTypes.string,
  link: PropTypes.string,
  active: PropTypes.bool,
  onClick: PropTypes.func,
  onHover: PropTypes.func,
  children: HIGChildValidator([SubmoduleComponent, ModuleCollapseComponent])
};

ModuleComponent.__docgenInfo = {
  props: {
    icon: {
      description: 'sets the icon of an Module'
    },

    title: {
      description: 'sets the title of an Module'
    },

    link: {
      description: 'sets the link of an Module'
    },

    active: {
      description: '[Boolean] Designates that the module is active'
    },

    onClick: {
      description: 'triggered when a link is clicked on'
    },

    onHover: {
      description: 'triggered when a link is hovered over'
    },

    children: {
      description: 'supports adding SubModule and Collapse'
    }
  }
};

ModuleComponent.Submodule = SubmoduleComponent;
ModuleComponent.ModuleCollapse = ModuleCollapseComponent;

export default ModuleComponent;
