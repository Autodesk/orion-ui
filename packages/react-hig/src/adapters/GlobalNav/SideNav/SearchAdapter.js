/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import createComponent from '../../createComponent';
import HIGElement from '../../../elements/HIGElement';

export class SearchAdapter extends HIGElement {
  componentDidMount() {
    this.commitUpdate(['clearIconVisible', this.props.clearIconVisible]);
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'placeholder': {
          this.hig.setPlaceholder(propValue);
          break;
        }
        case 'query': {
          this.hig.setQuery(propValue);
          break;
        }
        case 'clearIconVisible': {
          propValue ? this.hig.showClearIcon() : this.hig.hideClearIcon();
          break;
        }
        case 'onClearIconClick': {
          const dispose = this._disposeFunctions.get('onClearIconClickDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onClearIconClickDispose',
            this.hig.onClearIconClick(propValue)
          );
          break;
        }
        case 'onInput': {
          const dispose = this._disposeFunctions.get('onInputDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onInputDispose',
            this.hig.onInput(propValue)
          );
          break;
        }
        case 'onFocus': {
          const dispose = this._disposeFunctions.get('onFocusDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onFocusDispose',
            this.hig.onFocusIn(propValue)
          );
          break;
        }
        case 'onBlur': {
          const dispose = this._disposeFunctions.get('onBlurDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onBlurDispose',
            this.hig.onFocusOut(propValue)
          );
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const SearchComponent = createComponent(SearchAdapter);

SearchComponent.propTypes = {
  query: PropTypes.string,
  placeholder: PropTypes.string,
  clearIconVisible: PropTypes.bool,
  onClearIconClick: PropTypes.func,
  onInput: PropTypes.func,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func
};

SearchComponent.__docgenInfo = {
  props: {
    placeholder: {
      description: 'sets the placeholder'
    },

    query: {
      description: 'sets the search query'
    },

    clearIconVisible: {
      description: '{bool} show the clear input icon'
    },

    onClearIconClick: {
      description: '{func} triggered when clear icon is clicked'
    },

    onInput: {
      description: 'triggered when search field has something entered'
    },

    onFocus: {
      description: 'triggered when search field is in focus'
    },

    onBlur: {
      description: 'triggered when search field is not in focus'
    }
  }
};

export default SearchComponent;
