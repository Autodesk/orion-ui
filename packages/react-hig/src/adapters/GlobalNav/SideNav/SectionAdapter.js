/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';

import createComponent from '../../createComponent';
import HIGElement from '../../../elements/HIGElement';
import HIGNodeList from '../../../elements/HIGNodeList';
import HIGChildValidator from '../../../elements/HIGChildValidator';
import GroupComponent, { GroupAdapter } from './GroupAdapter';
import SectionCollapseComponent, {
  SectionCollapseAdapter
} from './SectionCollapseAdapter';

export class Section extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.groups = new HIGNodeList({
      type: GroupAdapter,
      HIGConstructor: this.hig.partials.Group,
      onAdd: (instance, beforeInstance) => {
        this.hig.addGroup(instance, beforeInstance);
      }
    });
  }

  componentDidMount() {
    this.groups.componentDidMount();

    if (this.collapse) {
      this.hig.addCollapse(this.collapse.hig);
      this.collapse.mount();
    }
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'headerLabel': {
          this.hig.setHeaderLabel(propValue);
          break;
        }
        case 'headerName': {
          this.hig.setHeaderName(propValue);
          break;
        }
        case 'children': {
          // No-op
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case GroupAdapter:
        return this.groups.createElement(ElementConstructor, props);
      case SectionCollapseAdapter:
        return new SectionCollapseAdapter(props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  insertBefore(instance, insertBeforeIndex) {
    this.appendChild(instance);
  }

  appendChild(instance) {
    if (instance instanceof GroupAdapter) {
      this.groups.insertBefore(instance); // calls internal _appendChild if no "before" component
    } else if (instance instanceof SectionCollapseAdapter) {
      if (this.collapse) {
        throw new Error('only one SectionCollapse is allowed');
      } else {
        this.collapse = instance;

        if (this.mounted) {
          instance.componentDidMount();
        }
      }
    } else {
      throw new Error('unknown type');
    }
  }

  removeChild(instance) {
    if (instance instanceof GroupAdapter) {
      this.groups.removeChild(instance);
    }

    if (instance instanceof SectionCollapseAdapter) {
      this.collpase = null;
      instance.unmount();
    }
  }
}

const SectionComponent = createComponent(Section);

SectionComponent.propTypes = {
  headerLabel: PropTypes.string,
  headerName: PropTypes.string,
  children: HIGChildValidator([GroupComponent, SectionCollapseComponent])
};

SectionComponent.__docgenInfo = {
  props: {
    headerLabel: {
      description: 'sets the label'
    },

    headerName: {
      description: 'sets the name'
    },

    children: {
      description: 'support adding Group and Collapse'
    }
  }
};

SectionComponent.Group = GroupComponent;
SectionComponent.SectionCollapse = SectionCollapseComponent;

export default SectionComponent;
