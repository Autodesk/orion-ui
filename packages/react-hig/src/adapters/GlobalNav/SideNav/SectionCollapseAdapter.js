/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import createComponent from '../../createComponent';
import HIGElement from '../../../elements/HIGElement';
import * as HIG from 'hig.web';

const HIGSectionCollapse = HIG.GlobalNav._partials.SideNav._partials.Section._partials.Collapse;

export class SectionCollapseAdapter extends HIGElement {
  constructor(props) {
    super(HIGSectionCollapse, props);
  }

  componentDidMount() {
    this.commitUpdate(['minimized', this.initialProps.minimized]);
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'minimized':
          propValue ? this.hig.minimize() : this.hig.maximize();
          break;
        case 'onClick': {
          const dispose = this._disposeFunctions.get('onClickDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onClickDispose',
            this.hig.onClick(propValue)
          );
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const SectionCollapseComponent = createComponent(SectionCollapseAdapter);

SectionCollapseComponent.propTypes = {
  minimized: PropTypes.bool,
  onClick: PropTypes.func
};

SectionCollapseComponent.__docgenInfo = {
  props: {
    minimized: {
      description: 'designate that collapse is in the minimized state'
    },

    onClick: {
      description: 'triggered when icon is clicked on'
    }
  }
};

export default SectionCollapseComponent;
