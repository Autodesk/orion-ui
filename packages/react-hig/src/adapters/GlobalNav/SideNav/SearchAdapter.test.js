/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import { mount } from 'enzyme';
import * as HIG from 'hig.web';
import React from 'react';

import GlobalNav from '../../../adapters/GlobalNav/GlobalNavAdapter';
import SideNav from './SideNavAdapter';
import Search from './SearchAdapter';
import Section from './SectionAdapter';
import SectionList from '../../../elements/components/GlobalNav/SectionList';
import Group from './GroupAdapter';
import Module from './ModuleAdapter';
import Submodule from './SubmoduleAdapter';

import SharedExamples
  from '../../../elements/components/GlobalNav/SharedExamples';

const Context = props => {
  return (
    <GlobalNav>
      <SideNav>
        <Search {...props} />
      </SideNav>
    </GlobalNav>
  );
};

function createOrionComponent(defaults) {
  const orionContainer = document.createElement('div');
  const orionWrapper = mount(<Context {...defaults} />, {
    attachTo: orionContainer
  });

  return { orionWrapper, orionContainer };
}

function createHigComponent(defaults = {}) {
  const higContainer = document.createElement('div');
  const globalNav = new HIG.GlobalNav();
  const sideNav = new globalNav.partials.SideNav();
  const search = new sideNav.partials.Search(defaults);

  globalNav.mount(higContainer);
  globalNav.addSideNav(sideNav);
  sideNav.addSearch(search);

  return { higComponent: search, higContainer };
}

describe('<Search>', () => {
  it('renders', () => {
    const defaults = {};
    const { higComponent, higContainer } = createHigComponent(defaults);
    const { orionContainer, orionWrapper } = createOrionComponent(defaults);

    expect(orionContainer.firstElementChild.outerHTML).toMatchSnapshot();

    expect(orionContainer.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });

  it('renders with initial props', () => {
    const defaults = {
      placeholder: 'Search a little search for me',
      query: 'Documents'
    };
    const { higComponent, higContainer } = createHigComponent(defaults);
    const { orionContainer, orionWrapper } = createOrionComponent(defaults);

    expect(orionContainer.firstElementChild.outerHTML).toMatchSnapshot();

    expect(orionContainer.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });

  it('renders with updated props', () => {
    const defaults = {};
    const nextProps = {
      placeholder: 'Search a little search for me',
      query: 'Documents',
      clearIconVisible: true
    };
    const { higComponent, higContainer } = createHigComponent(defaults);
    const { orionContainer, orionWrapper } = createOrionComponent(defaults);

    orionWrapper.setProps(nextProps);
    higComponent.setPlaceholder(nextProps.placeholder);
    higComponent.setQuery(nextProps.query);
    higComponent.showClearIcon();

    expect(orionContainer.firstElementChild.outerHTML).toMatchSnapshot();

    expect(orionContainer.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });

  it('warns when passed an unsupported property', () => {
    const warnSpy = jest.fn();
    const { orionContainer, orionWrapper } = createOrionComponent({});
    console.warn = warnSpy;

    orionWrapper.setProps({ realProp: false });

    expect(warnSpy).toHaveBeenCalled();
  });

  ['onClearIconClick', 'onInput', 'onFocus', 'onBlur'].forEach(eventName => {
    it(`handles ${eventName}`, () => {
      const warnSpy = jest.fn();
      const { orionContainer, orionWrapper } = createOrionComponent({});
      console.warn = warnSpy;

      orionWrapper.setProps({ [eventName]: () => {} });

      expect(warnSpy).not.toHaveBeenCalled();
    });
  });
});
