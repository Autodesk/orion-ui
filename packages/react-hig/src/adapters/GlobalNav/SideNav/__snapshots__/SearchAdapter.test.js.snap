// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search> renders 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__sidenav__scroll hig__global-nav__sidenav__scroll--search\\">
        <!--SECTION-->
        <div class=\\"hig__global-nav__sidenav__links\\">
            <!--LINK-->
        </div>
        <div class=\\"hig__global-nav__sidenav__copyright\\">© 2017 Autodesk, Inc.</div>
    </div>
    <div class=\\"hig__global-nav__side-nav__search\\">
    <div class=\\"hig__global-nav__side-nav__search__icon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"search-small\\"><title>hig/icon/search-small</title><path d=\\"M15.8702459,15.1631391 L18.9748737,18.267767 C19.1701359,18.4630291 19.1701359,18.7796116 18.9748737,18.9748737 C18.7796116,19.1701359 18.4630291,19.1701359 18.267767,18.9748737 L15.2403215,15.9474283 C14.0545271,17.2108151 12.3694256,18 10.5,18 C6.91014913,18 4,15.0898509 4,11.5 C4,7.91014913 6.91014913,5 10.5,5 C14.0898509,5 17,7.91014913 17,11.5 C17,12.8587306 16.5831026,14.1200897 15.8702459,15.1631391 Z M16,11.5 C16,8.46243388 13.5375661,6 10.5,6 C7.46243388,6 5,8.46243388 5,11.5 C5,14.5375661 7.46243388,17 10.5,17 C13.5375661,17 16,14.5375661 16,11.5 Z\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"></path></svg></div></div>
    <div class=\\"hig__global-nav__side-nav__search__inputholder\\"><input class=\\"hig__global-nav__side-nav__search__inputholder__input\\" type=\\"text\\" placeholder=\\"Search\\" value=\\"\\"></div>
    <div class=\\"hig__global-nav__side-nav__search__clear\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-small\\"><title>hig/icon/close-small</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><circle stroke=\\"#353636\\" cx=\\"12\\" cy=\\"12\\" r=\\"8\\"></circle><polygon fill=\\"#353636\\" transform=\\"rotate(45 12 12)\\" points=\\"16.5 11.4518414 12.5847458 11.4518414 12.5847458 7.5 11.4152542 7.5 11.4152542 11.4518414 7.5 11.4518414 7.5 12.5226629 11.4152542 12.5226629 11.4152542 16.5 12.5847458 16.5 12.5847458 12.5226629 16.5 12.5226629\\"></polygon></g></svg></div></div>
</div><!--SEARCH-->
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Search> renders with initial props 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__sidenav__scroll hig__global-nav__sidenav__scroll--search\\">
        <!--SECTION-->
        <div class=\\"hig__global-nav__sidenav__links\\">
            <!--LINK-->
        </div>
        <div class=\\"hig__global-nav__sidenav__copyright\\">© 2017 Autodesk, Inc.</div>
    </div>
    <div class=\\"hig__global-nav__side-nav__search\\">
    <div class=\\"hig__global-nav__side-nav__search__icon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"search-small\\"><title>hig/icon/search-small</title><path d=\\"M15.8702459,15.1631391 L18.9748737,18.267767 C19.1701359,18.4630291 19.1701359,18.7796116 18.9748737,18.9748737 C18.7796116,19.1701359 18.4630291,19.1701359 18.267767,18.9748737 L15.2403215,15.9474283 C14.0545271,17.2108151 12.3694256,18 10.5,18 C6.91014913,18 4,15.0898509 4,11.5 C4,7.91014913 6.91014913,5 10.5,5 C14.0898509,5 17,7.91014913 17,11.5 C17,12.8587306 16.5831026,14.1200897 15.8702459,15.1631391 Z M16,11.5 C16,8.46243388 13.5375661,6 10.5,6 C7.46243388,6 5,8.46243388 5,11.5 C5,14.5375661 7.46243388,17 10.5,17 C13.5375661,17 16,14.5375661 16,11.5 Z\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"></path></svg></div></div>
    <div class=\\"hig__global-nav__side-nav__search__inputholder\\"><input class=\\"hig__global-nav__side-nav__search__inputholder__input\\" type=\\"text\\" placeholder=\\"Search a little search for me\\" value=\\"Documents\\"></div>
    <div class=\\"hig__global-nav__side-nav__search__clear\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-small\\"><title>hig/icon/close-small</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><circle stroke=\\"#353636\\" cx=\\"12\\" cy=\\"12\\" r=\\"8\\"></circle><polygon fill=\\"#353636\\" transform=\\"rotate(45 12 12)\\" points=\\"16.5 11.4518414 12.5847458 11.4518414 12.5847458 7.5 11.4152542 7.5 11.4152542 11.4518414 7.5 11.4518414 7.5 12.5226629 11.4152542 12.5226629 11.4152542 16.5 12.5847458 16.5 12.5847458 12.5226629 16.5 12.5226629\\"></polygon></g></svg></div></div>
</div><!--SEARCH-->
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Search> renders with updated props 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__sidenav__scroll hig__global-nav__sidenav__scroll--search\\">
        <!--SECTION-->
        <div class=\\"hig__global-nav__sidenav__links\\">
            <!--LINK-->
        </div>
        <div class=\\"hig__global-nav__sidenav__copyright\\">© 2017 Autodesk, Inc.</div>
    </div>
    <div class=\\"hig__global-nav__side-nav__search\\">
    <div class=\\"hig__global-nav__side-nav__search__icon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"search-small\\"><title>hig/icon/search-small</title><path d=\\"M15.8702459,15.1631391 L18.9748737,18.267767 C19.1701359,18.4630291 19.1701359,18.7796116 18.9748737,18.9748737 C18.7796116,19.1701359 18.4630291,19.1701359 18.267767,18.9748737 L15.2403215,15.9474283 C14.0545271,17.2108151 12.3694256,18 10.5,18 C6.91014913,18 4,15.0898509 4,11.5 C4,7.91014913 6.91014913,5 10.5,5 C14.0898509,5 17,7.91014913 17,11.5 C17,12.8587306 16.5831026,14.1200897 15.8702459,15.1631391 Z M16,11.5 C16,8.46243388 13.5375661,6 10.5,6 C7.46243388,6 5,8.46243388 5,11.5 C5,14.5375661 7.46243388,17 10.5,17 C13.5375661,17 16,14.5375661 16,11.5 Z\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"></path></svg></div></div>
    <div class=\\"hig__global-nav__side-nav__search__inputholder\\"><input class=\\"hig__global-nav__side-nav__search__inputholder__input\\" type=\\"text\\" placeholder=\\"Search a little search for me\\" value=\\"\\"></div>
    <div class=\\"hig__global-nav__side-nav__search__clear hig__global-nav__side-nav__search__clear--show\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-small\\"><title>hig/icon/close-small</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><circle stroke=\\"#353636\\" cx=\\"12\\" cy=\\"12\\" r=\\"8\\"></circle><polygon fill=\\"#353636\\" transform=\\"rotate(45 12 12)\\" points=\\"16.5 11.4518414 12.5847458 11.4518414 12.5847458 7.5 11.4152542 7.5 11.4152542 11.4518414 7.5 11.4518414 7.5 12.5226629 11.4152542 12.5226629 11.4152542 16.5 12.5847458 16.5 12.5847458 12.5226629 16.5 12.5226629\\"></polygon></g></svg></div></div>
</div><!--SEARCH-->
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
